cpp/src_prims/selection/knn.h

```diff
@@ -406,7 +406,7 @@ __global__ void class_vote_kernel(OutType *out, const ProbaType *class_proba,
   __syncthreads();
 
   if (row >= n_samples) return;
-  int cur_max = -1;
+  ProbaType cur_max = -1;
```

Comment:
why is cur_count float instead of ProbaType?

Reply:
Actually, I think this should always be a float, since it’s a probability and the use of double will not provide any benefit. The  ProbaType was just created to make the typing extensible (and save future refractors).